plugins/experimental/header_freq/header_freq.cc

```diff
@@ -138,7 +164,16 @@ count_all_headers(TSMBuffer &bufp, TSMLoc &hdr_loc, std::map<std::string, unsign
       c = tolower(c);
     }
 
-    ++map[str];
+    if (map.find(str) == map.end()) {
+      std::unique_lock<std::shared_mutex> lock{map_mutex};
```

Comment:
You have to lock during search because updating the map can break search.

Reply:
Yeah, good point. I should grab the read lock before then and upgrade if the entry needs to be added.

Reply:
Updated.